src/libhoney.js

```diff
@@ -493,3 +496,23 @@ function getAndInitTransmission(transmission, options) {
     }
   }
 }
+
+function concatWithMaxLimit(arr1, arr2, limit) {
+  // if queue is full or somehow over the max
+  if (arr1.length >= limit) {
+    //return up to the max length
+    return arr1.slice(0, limit);
```

Comment:
@toshok I really like this return using the limit as a slice. Would it make sense to reverse this and use this as a fallback instead of the first check? Not sure which scenario is most common. This early bail if you're at max makes sense for when the max is 0, so maybe a good move to keep it here.